orchestration/templates/download-ftp-file.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: download-ftp-file
+spec:
+  templates:
+    - name: main
+      inputs:
+        parameters:
+          - name: ftp-site
+          - name: ftp-path
+          - name: local-path
+          - name: pvc-name
+      volumes:
+        - name: state
+          persistentVolumeClaim:
+            claimName: '{{ "{{inputs.parameters.pvc-name}}" }}'
+      container:
+        image: cirrusci/wget
+        command: [wget]
+        args:
+          # Restart from any state
+          - --continue
+          # Reset the download connection if it goes idle for > 30 seconds.
+          - --read-timeout=30
+          - --output-document=/state/{{ "{{inputs.parameters.local-path}}" }}
+          - ftp://{{ "{{inputs.parameters.ftp-site}}" }}/{{ "{{inputs.parameters.ftp-path}}" }}
+        volumeMounts:
+          - name: state
+            mountPath: /state
+        resources:
+          requests:
+            memory: 512Mi
+            cpu: 1000m
+          limits:
+            memory: 512Mi
+            cpu: 1000m
+      retryStrategy:
+        limit: 32
```

Comment:
I chose to extract templates that felt truly reusable into their own files. Eventually it could make sense to move these to `monster-helm` as a sort of "utility chart" of templates.